x-pack/plugins/session_view/public/components/session_view/index.tsx

```diff
@@ -48,12 +56,13 @@ export const SessionView = ({ sessionEntityId, height, jumpToEvent }: SessionVie
     hasPreviousPage,
   } = useFetchSessionViewProcessEvents(sessionEntityId, jumpToEvent);
 
-  const hasData = data && data.pages.length > 0 && data.pages[0].events.length > 0;
-  const renderIsLoading = isFetching && !data;
+  const alertsQuery = useFetchSessionViewAlerts(sessionEntityId);
+  const { data: alerts, error: alertsError, isFetching: alertsFetching } = alertsQuery;
+
+  const hasData = alerts && data && data.pages?.[0].events.length > 0;
```

Comment:
is `alerts` here in case fetching alerts ran into error? Since I imagine if we don't have alerts, it will just be a truthy `[]`. If it is for errors, would the alertsError be enough for error handling?

Reply:
It's to avoid passing an undefined to ProcessTree as the prop expects an array.

Reply:
alerts is initially undefined until the react-query fetch completes.